src/Discord.Net.Core/Entities/Messages/Embed.cs

```diff
@@ -56,6 +57,8 @@ internal Embed(string type)
             Fields = fields;
         }
 
+        public int Length => Title?.Length + Author?.Name?.Length + Description?.Length + Footer?.Text?.Length + Fields.Sum(f => f.Name.Length + f.Value.ToString().Length) ?? 0;
```

Comment:
This assumes only text properties count toward the limit; if URLs count then obviously this needs an update :)